packages/client/src/schema/pagination.ts

```diff
@@ -104,11 +104,50 @@ export function isCursorPaginationOptions(
 }
 
 export class RecordArray<Result extends XataRecord> extends Array<Result> {
+  constructor(overrideRecords?: Result[]);
+  constructor(...args: any[]) {
+    super(...RecordArray.parseConstructorParams(...args));
+  }
+
+  static parseConstructorParams(...args: any[]) {
+    // new <T>(arrayLength: number): T[]
```

Comment:
What do the comments mean? 

Like someone could call `new RecordArray<SomeType>(1)`or `new RecordArray<SomeType>(["", ""])`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Array#parameters